src/main/java/carpet/script/api/WorldAccess.java

```diff
@@ -1626,23 +1634,72 @@ else if (ticket == TicketType.POST_TELEPORT) // post teleport
-            ret.put(new StringValue(noise), new NumericValue(noiseValue));
+        ChunkGenerator chunkGenerator = level.getChunkSource().getGenerator();
+        if (!(chunkGenerator instanceof NoiseBasedChunkGenerator generator)) {
+            return MapValue.wrap(result);
+        }
+
+        RandomState randomState = RandomState.create(
+            generator.generatorSettings().value(),
+            level.registryAccess().lookupOrThrow(Registries.NOISE), level.getSeed()
+        );
+
+        DensityFunction.Visitor visitor = ((RandomStateVisitorAccessor) (Object) randomState).getVisitor();
+        DensityFunction.SinglePointContext singlePointContext = new DensityFunction.SinglePointContext(pos.getX(), pos.getY(), pos.getZ());
+        NoiseRouter router = generator.generatorSettings().value().noiseRouter();
+
+        for (String densityFunctionQuery : densityFunctionQueries) {
+            DensityFunction densityFunction = switch (densityFunctionQuery) {
+                case "barrier_noise" -> router.barrierNoise();
+                case "fluid_level_floodedness_noise" -> router.fluidLevelFloodednessNoise();
+                case "fluid_level_spread_noise" -> router.fluidLevelSpreadNoise();
+                case "lava_noise" -> router.lavaNoise();
+                case "temperature" -> router.temperature();
+                case "vegetation" -> router.vegetation();
+                case "continents" -> router.continents();
+                case "erosion" -> router.erosion();
+                case "depth" -> router.depth();
+                case "ridges" -> router.ridges();
+                case "initial_density_without_jaggedness" -> router.initialDensityWithoutJaggedness();
+                case "final_density" -> router.finalDensity();
+                case "vein_toggle" -> router.veinToggle();
+                case "vein_ridged" -> router.veinRidged();
+                case "vein_gap" -> router.veinGap();
+                default -> {
+                    ResourceLocation densityFunctionKey = InputValidator.identifierOf(densityFunctionQuery);
+
+                    if (densityFunctionRegistry.containsKey(densityFunctionKey)) {
+                        yield densityFunctionRegistry.get(densityFunctionKey);
+                    }
+
+                    throw new InternalExpressionException(
+                        "Density function '" + densityFunctionQuery + "' doesn't exist. " +
+                        "Please check the spelling or use with full 'namespace:value' if its custom defined."
+                    );
+                }
+            };
+
+            if (densityFunction == null) {
+                throw new InternalExpressionException(
+                    "Somehow density function was not found in the whole registry and error checking didn't trigger." +
+                    "Ideally this should never reach here."
+                );
+            }
```

Comment:
Can this actually ever happen? If it shouldn't I'd remove it or throw a non-scarpet exception (like IllegalState/Argument)

Reply:
No. This should never happen. I did it just to make one less warning "densityFuction could be null"

Reply:
```suggestion
            if (densityFunction == null) // IDE thinks may be null as Registry.get is Nullable, but we checked it
                throw new IllegalStateException("density function can't be null here"); 
```